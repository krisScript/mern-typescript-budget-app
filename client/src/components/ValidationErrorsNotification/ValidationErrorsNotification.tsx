import React, { FunctionComponent } from 'react';
interface ValidationErrorsNotificationProps {
  validationErrorMessages: string[];
}
const ValidationErrorsNotification: FunctionComponent<
  ValidationErrorsNotificationProps
> = ({ validationErrorMessages }): JSX.Element => {
  return (
    <>
      {validationErrorMessages.length > 0 ? (
        <div data-testid="error-list" className="notification is-danger">
          {validationErrorMessages.map(
            (errorMessage: string): JSX.Element => (
              <p key={errorMessage}>{errorMessage}</p>
            ),
          )}
        </div>
      ) : (
        ''
      )}
    </>
  );
};
export default ValidationErrorsNotification;
