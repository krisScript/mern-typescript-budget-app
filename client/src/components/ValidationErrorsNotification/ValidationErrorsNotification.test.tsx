import React from 'react';
import { render, cleanup, fireEvent, getByTestId } from 'react-testing-library';
import { BrowserRouter } from 'react-router-dom';
import ValidationErrorsNotification from './ValidationErrorsNotification';

describe('<ValidationErrorsNotification />', () => {
  const validationErrorsMessages: string[] = [];
  const { container, rerender, queryByTestId, getByTestId } = render(
    <ValidationErrorsNotification
      validationErrorMessages={validationErrorsMessages}
    />,
  );

  it('snapshot', () => {
    expect(container).toMatchSnapshot();
  });
  it('errors-list', async () => {
    const errorsList = queryByTestId('error-list');
    expect(errorsList).toBeNull();
    const newErrorsList = ['Email is already taken!', 'Wrong Password!'];
    rerender(
      <ValidationErrorsNotification validationErrorMessages={newErrorsList} />,
    );
    const rerenderedErrorsList = getByTestId('error-list');
    expect(rerenderedErrorsList.childElementCount).toBe(2);
  });
});
