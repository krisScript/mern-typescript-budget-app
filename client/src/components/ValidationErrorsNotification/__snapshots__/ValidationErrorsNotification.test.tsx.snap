// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ValidationErrorsNotification /> snapshot 1`] = `
<div>
  
</div>
`;
